mono/mini/debugger-agent.c

```diff
@@ -249,6 +251,9 @@ typedef struct {
 
 	/* The currently unloading appdomain */
 	MonoDomain *domain_unloading;
+
+	// The state that the debugger expects the thread to be in
+	MonoDebuggerThreadState thread_state;
 } DebuggerTlsData;
```

Comment:
To avoid having `mono_debugger_tls_thread_id (gpointer debuggerTlsData)` with `debuggerTlsData` being a `gpointer`, you can simply define `DebuggerTlsData` as an opaque struct in `debugger-agent.h` like `typedef struct _ DebuggerTlsData DebuggerTlsData;` and have `struct _DebuggerTlsData` at https://github.com/mono/mono/pull/8865/files#diff-cad67a3642d7f7e00161e3e81c29f25eL168. This will allow you to use `DebuggerTlsData*` anywhere you want, without having to expose its definition.